pkg/cli/plugin_cmd.go

```diff
@@ -172,12 +233,22 @@ func getHelpArguments() []string {
 
 // setupPluginEnv prepares some extra environment variables
 // that communicate certain information to plugins.
-func setupPluginEnv() {
+func setupPluginEnv(srcHierarchy, dstHierarchy []string) {
 	env := make(map[string]string, 10)
 
 	// The location of the tanzu binary
 	env["TANZU_BIN"] = os.Args[0]
 
+	// Information about if command invocation is via a mapped command
+	numParts := len(dstHierarchy)
+	if numParts > 0 {
+		env["TANZU_CLI_INVOKED_COMMAND"] = dstHierarchy[numParts-1]
+		if numParts > 1 {
+			env["TANZU_CLI_INVOKED_GROUP"] = strings.Join(dstHierarchy[:numParts-1], "")
```

Comment:
Should the separator be `" "` instead of `""`?
When testing I mapped `builder cli` to `plugin source mycli` and saw: 
```
$ tz plugin source mycli -h
Build CLIs

Usage:
  tanzu pluginsource mycli
```
which is missing the space

Reply:
nice catch, and good one on getting creative to test mapping 2 levels deep!
fixed and updated the test as well.